File: shop_service/src/main/java/com/luckytree/shop_service/shop/adapter/feign/MemberFeignClient.java
package com.luckytree.shop_service.shop.adapter.feign;


import com.luckytree.shop_service.common.config.FeignClientConfig;
import com.luckytree.shop_service.shop.adapter.data.MemberFeignRequestDto;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.PostMapping;

@FeignClient(name = "member-service", configuration = FeignClientConfig.class)
public interface MemberFeignClient {

    @PostMapping("/v1/bookmarks")
    void saveBookmark(MemberFeignRequestDto memberFeignRequestDto);
    @DeleteMapping("/v1/bookmarks/{memberId}")
    void deleteBookmark(@PathVariable long memberId, @RequestParam("shopId") long shopId);
}

File: auth/src/main/kotlin/com/example/auth/client/feign/member/MemberFeignClient.kt
package com.example.auth.client.feign.member

import com.example.auth.client.feign.member.data.MemberFeignResponse
import com.example.auth.config.InternalFeignConfig
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam

@FeignClient(value = "member", configuration = [InternalFeignConfig::class])
interface MemberFeignClient {
    @GetMapping("member/v1/members")
    fun findByIdAndEmail(@RequestParam id: Long, @RequestParam email: String): MemberFeignResponse
}
